rbac/api/utils.py

```diff
@@ -1,15 +1,49 @@
 """Helper utilities for api module."""
 
+import logging
 from django.db import transaction
+from django.shortcuts import get_object_or_404
 from management.utils import account_id_for_tenant
 
 from api.models import Tenant
+from management.role.model import BindingMapping
+from management.tenant_mapping.model import TenantMapping
+from management.workspace.model import Workspace
 
 
+RESOURCE_MODEL_MAPPING = {
+    "workspace": Workspace,
+    "mapping": TenantMapping,
+    "binding": BindingMapping,
+}
+logger = logging.getLogger(__name__)
+
 def populate_tenant_account_id():
     """Populate tenant's account id's."""
     tenants = Tenant.objects.filter(account_id__isnull=True).exclude(tenant_name="public")
     for tenant in tenants:
         with transaction.atomic():
             tenant.account_id = account_id_for_tenant(tenant)
             tenant.save()
+
+
+def migration_resource_deletion(resource, org_id):
+    resource_objs = RESOURCE_MODEL_MAPPING[resource].objects.all()
+    if org_id:
+        tenant = get_object_or_404(Tenant, org_id=org_id)
+        if resource == "binding":
+            resource_objs = resource_objs.filter(role__tenant=tenant)
+        else:
+            resource_objs = resource_objs.filter(tenant=tenant)
+    else:
+        if resource == "binding":
+            public_tenant = Tenant.objects.get(tenant_name="public")
+            resource_objs = resource_objs.exclude(role__tenant=public_tenant)
```

Comment:
Why are these excluded in this case? It would mean we still leave bindings around if they have been bound with system roles. Do we want that?